### DISCUSSION TITLE
Using iteration as list results in unexpected behavior of data objects

### DISCUSSION THREAD

**Discussion by jennysjaarda** [ropensci/targets#511] [General]
Sorry for the many threads recently. I'm not sure if this behavior is expected so just wanted to bring to your attention / the other users. I'm trying to use columns from a tibble/df as inputs into a function and also taking advantage of iteration as list and came across this. The reason I want to use the `list` iteration is because of the way the third target is set up. 

```
# _targets.R

library(targets)
library(tarchetypes)


tar_option_set(error = "workspace")
tar_option_set(packages = c("biglm", "tidyverse"))

list(
  tar_target(random_vec, tibble(i = 1:10, j=11:20) , iteration = "list"), #works as expected if I just set middle arguement as 1:10
  tar_target(rnorm_random_vec, rnorm(random_vec$i), pattern = map(random_vec), iteration = "list"),
  tar_target(count_random_vec, length(rnorm_random_vec), pattern = map(rnorm_random_vec), iteration = "list")
)

```

Running `tar_make()` errors on `rnorm_random_vec` target. 

```
Error : $ operator is invalid for atomic vectors
Error: callr subprocess failed: $ operator is invalid for atomic vectors
Visit https://books.ropensci.org/targets/debugging.html for debugging advice.
```

If I enter the workspace of the errored out target I get an expected result of `random_vec` that differs from `tar_read(random_vec)`. 

```
> tar_workspace(rnorm_random_vec_05de47d3)
> random_vec
 [1]  1  2  3  4  5  6  7  8  9 10

> tar_read(random_vec)
# A tibble: 10 x 2
       i     j
   <int> <int>
 1     1    11
 2     2    12
 3     3    13
 4     4    14
 5     5    15
 6     6    16
 7     7    17
 8     8    18
 9     9    19
10    10    20

```

**Comment by wlandau** [comment]
If you write `iteration = "list"` for `random_vec`, then downstream dynamic targets will iterate over it as a list: first `random_vec[[1]]` and then `random_vec[[2]]`. In other words, this behavior iterates over the columns, not the rows. To iterate over the rows instead, you could stick with the default `iteration = "vector"` or write `purrr::transpose(tibble(i = 1:10, j=11:20))` with list iteration. Both worked when I tried them.
